sdk/eventhub/azure-eventhubs/azure/eventhub/producer.py

```diff
@@ -327,13 +349,15 @@ def send(self, event_data, partition_key=None):
         self._check_closed()
         if isinstance(event_data, EventData):
             if partition_key:
-                event_data._set_partition_key(partition_key)
+                event_data._set_partition_key(partition_key)  # pylint: disable=protected-access
             wrapper_event_data = event_data
         else:
-            event_data_with_pk = self._set_partition_key(event_data, partition_key)
-            wrapper_event_data = _BatchSendEventData(
-                event_data_with_pk,
-                partition_key=partition_key) if partition_key else _BatchSendEventData(event_data)
+            if isinstance(event_data, EventDataBatch):  # The partition_key in the param will be omitted.
+                wrapper_event_data = event_data
```

Comment:
Same comment as above regarding raising an error if partition_key is going to be ignored.

Reply:
Code updated, an EventDataError will be raised if the keys don't match.